### PR TITLE
Proposal for authorized guards

### PR DESCRIPTION
**Pull request by eutj**
Just a proposal. Don't approve!

### CODE REVIEW

File: new-frontend/src/app/guards/chosenCollaboration.guard.ts [hunk 1]
```diff
@@ -2,16 +2,23 @@ import { inject } from '@angular/core';
 import { CanActivateFn, Router } from '@angular/router';
 import { routePaths } from '../routes';
 import { CHOSEN_COLLABORATION } from '../models/constants/sessionStorage';
+import { ResourceType, OperationType } from '../models/api/rule.model';
+import { PermissionService } from '../services/permission.service';
+import { CollaborationService } from '../services/collaboration.service';
 
-export function chosenCollaborationGuard(): CanActivateFn {
+export function chosenCollaborationGuard(resource?: ResourceType, operation?: OperationType): CanActivateFn {
```
Comment (richard-elastique) [inline new-frontend/src/app/guards/chosenCollaboration.guard.ts:9]: This guard has only one purpose, checking if there is a chosenCollaboration the user is allowed to use. It does not need resource and operation as a parameter, as these values will always be the same.

File: new-frontend/src/app/guards/chosenCollaboration.guard.ts [hunk 2]
```diff
@@ -2,16 +2,23 @@ import { inject } from '@angular/core';
...
+    const collaborationService: CollaborationService = inject(CollaborationService);
 
     const chosenCollaboration = sessionStorage.getItem(CHOSEN_COLLABORATION);
     if (!chosenCollaboration) {
       router.navigate([routePaths.start]);
       return false;
     }
-    return true;
+
+    const collaboration = await collaborationService.getCollaboration(chosenCollaboration);
```
Comment (richard-elastique) [inline new-frontend/src/app/guards/chosenCollaboration.guard.ts:21]: Use the already loaded collaboration data in the ChosenCollaborationService. Loading data in a guard will cause this data to be loaded on every page view that uses this guard and slows the application unnecessarily. Especially if the needed data is already loaded in a service.

**Review by richard-elastique — CHANGES_REQUESTED** [review]

File: new-frontend/src/app/guards/chosenCollaboration.guard.ts (hunk 1 above)
Comment (bartvanb) [reply new-frontend/src/app/guards/chosenCollaboration.guard.ts:9]: I don't think that's true: this collaboration guard can be used for different pages. In the PR, it is used in to check if someone is allowed to access a page where you can create a task for that collaboration:
```
        path: routerConfig.taskCreate,
        component: TaskCreateComponent,
        canActivate: [authenticationGuard(), chosenCollaborationGuard(ResourceType.TASK, OperationType.CREATE)]
```
This guard may also be used to access a page where you view all the tasks of that collaboration, or where you edit/delete the collaboration itself

File: new-frontend/src/app/guards/chosenCollaboration.guard.ts [hunk 3]
```diff
@@ -2,16 +2,23 @@ import { inject } from '@angular/core';
...
+    return resource != null && operation != null ? permissionService.isAllowedForCollab(resource, operation, collaboration) : true;
```
Comment (bartvanb) [inline new-frontend/src/app/guards/chosenCollaboration.guard.ts:22]: Should the `true` at the end not be a `false`?

**Comment by eutj** [comment]
Ultimately, the optional parameters will be replaced by required parameters. For now, returning true won’t break existing functionality.

File: new-frontend/src/app/guards/chosenCollaboration.guard.ts (hunk 3 above)
Comment (eutj) [reply new-frontend/src/app/guards/chosenCollaboration.guard.ts:22]: Ultimately, the optional parameters will be replaced by required parameters. For now, returning true won’t break existing functionality.

File: new-frontend/src/app/guards/chosenCollaboration.guard.ts (hunk 1 above)
Comment (richard-elastique) [reply new-frontend/src/app/guards/chosenCollaboration.guard.ts:9]: I would expect the chosenCollaborationGuard to only check if there is a chosen collaboration. I than would expect a new authorizationGuard to check if the user has the right roles. So the chosenCollaborationGuard can be used on pages that need a chosen collaboration (for example the whole admin section doesn't that) and the authorizationGuard can be used for all pages that need a specific role (or multiple).

**Pull request closed without merging**